Sources/Rendering/OpenGL/CustomPostProcessingPass/index.js

```diff
@@ -0,0 +1,317 @@
+      'u_kernelWeight',
+      publicAPI.computeKernelWeight(model.kernel)
+    );
+
+    // render quad
+    gl.drawArrays(gl.TRIANGLES, 0, model.tris.getCABO().getElementCount());
+    tex.deactivate();
+  };
+
+  publicAPI.getFragmentShaderCode = (kernelDimension) => {
+    // generate new shader code
+    const kernelLength = kernelDimension * kernelDimension;
+    let shaderCode = [
+      '//VTK::System::Dec',
+      '//VTK::Output::Dec',
+      'uniform sampler2D u_image;',
+      'uniform vec2 u_textureSize;',
+      `uniform float u_kernel[${kernelLength}];`,
+      'uniform float u_kernelWeight;',
+      'varying vec2 tcoord;',
+      'void main(){',
+      '    vec2 onePixel = vec2(1.0, 1.0) / u_textureSize;',
+      '    vec4 colorSum =\n',
+    ].join('\n');
+
+    const halfDim = Math.floor(kernelDimension / 2);
+
+    // generate sum per pixel
+    let i = 0;
+    for (let y = -halfDim; y <= halfDim; ++y) {
+      for (let x = -halfDim; x <= halfDim; ++x) {
+        shaderCode += `        texture2D(u_image, tcoord + onePixel * vec2(${x}, ${y})) * u_kernel[${i}]`;
+        ++i;
+
+        if (i !== kernelLength) {
+          shaderCode += ' +\n';
+        }
+      }
+    }
+
+    // finish code
+    shaderCode += [
+      ';',
+      '    gl_FragData[0] = vec4((colorSum / u_kernelWeight).rgb, 1);',
+      '}',
+    ].join('\n');
+
+    return shaderCode;
+  };
+
+  // build vertices etc
+  publicAPI.buildVertexBuffer = () => {
+    // 4 coner points in clipping space in order (x, y, z) where z is always set to -1
+    const ptsArray = new Float32Array([
+      -1,
+      -1,
+      -1,
+      1,
+      -1,
+      -1,
+      -1,
+      1,
+      -1,
+      1,
+      1,
+      -1,
+    ]);
+
+    // 4 corresponding corner points in texture space in order (x, y)
+    const tcoordArray = new Float32Array([0, 0, 1, 0, 0, 1, 1, 1]);
+
+    // a square defined as cell relation ship in order (cell_size, v1, v2, v3, v4)
+    const cellArray = new Uint16Array([4, 0, 1, 3, 2]);
+
+    const points = vtkDataArray.newInstance({
+      numberOfComponents: 3,
+      values: ptsArray,
+    });
+    points.setName('points');
+    const tcoords = vtkDataArray.newInstance({
+      numberOfComponents: 2,
+      values: tcoordArray,
+    });
+    tcoords.setName('tcoords');
+    const cells = vtkDataArray.newInstance({
+      numberOfComponents: 1,
+      values: cellArray,
+    });
+    model.tris.getCABO().createVBO(cells, 'polys', Representation.SURFACE, {
+      points,
+      tcoords,
+      cellOffset: 0,
+    });
+
+    model.VBOBuildTime.modified();
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  framebuffer: null,
+  convolutionShader: null,
+  tris: null,
+  kernel: [0, 0, 0, 0, 1, 0, 0, 0, 0],
+  oldKernelDimension: 3,
+  kernelDimension: 3,
```

Comment:
`kernelSize`

Reply:
Strictly speaking kernelSize would be e.g. 3x3. Therefore, we would rather keep the kernelDimension since it indicates one side of the square matrix. Do you have a better idea to make it more intuitive?

Reply:
I think you're right, dimension seems fine.